src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -294,6 +311,12 @@ void Instance::OnResult(Status commissioningError, CharSpan errorText, int32_t i
     response.debugText        = errorText;
     response.errorValue       = interfaceStatus;
     commandHandle->AddResponseData(mPath, response);
+
+    mLastNetworkIDLen = mConnectingNetworkIDLen;
+    memcpy(mLastNetworkID, mConnectingNetworkID, sizeof(mConnectingNetworkID));
```

Comment:
I think we should only copy `mLastNetworkIDLen` bytes, not `sizeof(mConnectingNetworkID)`